src/ray/util/util.h

```diff
@@ -35,4 +38,33 @@ inline ray::Status boost_to_ray_status(const boost::system::error_code &error) {
   }
 }
 
+template <class Shutdown>
+class InitShutdownWrapper {
+ public:
+  /// Create an instance of InitShutdownWrapper which will call shutdown
+  /// function when it is out of scope.
+  ///
+  /// \param init_func The init function.
+  /// \param shuntdown_func The shutdown function.
+  /// \param args The auguments for the init function.
+  template <class Init, class... Args>
+  InitShutdownWrapper(Init init_func, Shutdown shuntdown_func, Args &&... args)
+      : shutdown_(shuntdown_func) {
+    init_func(args...);
+  }
+
+  /// Destructor of InitShutdownWrapper which will call the shutdown function.
+  ~InitShutdownWrapper() {
+    if (shutdown_ != nullptr) {
+      shutdown_();
+    }
+  }
+
+ private:
+  Shutdown shutdown_;
+};
+
+// Most of the shutdown function is the type of void (*)().
+using DefaultInitShutdown = InitShutdownWrapper<void (*)()>;
```

Comment:
Other people might think it's fine, but I'd remove template and `using` and just implement the wrapper for this case.  Less complexity for the win.